airbyte-webapp/src/packages/cloud/views/auth/SignupPage/components/SignupForm.tsx

```diff
@@ -75,40 +81,78 @@ export const CompanyNameField: React.FC = () => {
+    }
+    return null;
+  };
   return (
     <Field name="email">
       {({ field, meta }: FieldProps<string>) => (
-        <LabeledInput
-          {...field}
-          label={label || <FormattedMessage id="login.yourEmail" />}
-          placeholder={formatMessage({
-            id: "login.yourEmail.placeholder",
-          })}
-          type="text"
-          error={!!meta.error && meta.touched}
-          message={meta.touched && meta.error && formatMessage({ id: meta.error })}
-        />
+        <>
+          {" "}
+          <LabeledInput
+            {...field}
+            label={label || <FormattedMessage id="login.yourEmail" />}
+            placeholder={formatMessage({
+              id: "login.yourEmail.placeholder",
+            })}
+            type="text"
+            error={(!!meta.error && meta.touched) || (meta.touched && !isCorporateEmail(field.value))}
+            message={getMessage({ touched: meta.touched, error: meta.error, value: field.value })}
+          />
+        </>
       )}
     </Field>
   );
 };
 
 export const PasswordField: React.FC<{ label?: React.ReactNode }> = ({ label }) => {
   const { formatMessage } = useIntl();
+  const isSimpleLeftSide = useExperiment("authPage.signup.simplifyLeftSide", false);
 
   return (
     <Field name="password">
       {({ field, meta }: FieldProps<string>) => (
-        <LabeledInput
-          {...field}
-          label={label || <FormattedMessage id="login.password" />}
-          placeholder={formatMessage({
-            id: "login.password.placeholder",
-          })}
-          type="password"
-          error={!!meta.error && meta.touched}
-          message={meta.touched && meta.error && formatMessage({ id: meta.error })}
-        />
+        <>
+          <LabeledInput
+            {...field}
+            label={label || <FormattedMessage id="login.password" />}
+            placeholder={formatMessage({
+              id: "login.password.placeholder",
+            })}
+            type="password"
+            error={!!meta.error && meta.touched}
+            message={meta.touched && meta.error && formatMessage({ id: meta.error })}
+          />
+          {isSimpleLeftSide && (
+            <FlexContainer gap="sm" alignItems="center" className={styles.passwordCheckContainer}>
+              <FontAwesomeIcon
+                icon={Boolean(meta.error) && meta.touched ? faXmarkCircle : faCheckCircle}
+                className={classNames(styles.checkIcon, {
+                  [styles.error]: Boolean(meta.error) && meta.touched,
+                  [styles.valid]: meta.touched && !meta.error,
+                })}
+              />
+
+              <Text
+                size="sm"
+                className={classNames({
+                  [styles.error]: Boolean(meta.error) && meta.touched,
+                })}
+              >
+                <FormattedMessage id="signup.password.minLength" />
+              </Text>
+            </FlexContainer>
+          )}
```

Comment:
We already have an issue out at https://github.com/airbytehq/airbyte/issues/21702 to improve that error handling. I think we don't need to add the error message improvements before that. But if we do so, it should imho not behind an experimentation flag, but just be added to the UI.

Reply:
I think for things we clearly know are better (show users form validation) there's no purpose in putting them behind an experiment, and will just clutter the code and thus maintainability. But I won't block this PR on this opinion, but a more general thought for you and @nataliekwong to discuss moving forward.

Reply:
I agree on having things that we know are better for users shouldn't usually live under an experiment or FF, but in this case, my reasoning was:
- I have the FF anyway
- it's not going to live long in the code 
But I will remove it along with other changes.